src/App.tsx

```diff
@@ -1,73 +1,107 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Autocomplete } from './components/Autocomplete';
+import { Person } from './types/Person';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+const debounce = (callback: Function, delay: number = 300) => {
+  let timerId = 0;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+    timerId = window.setTimeout(() => callback(...args), delay);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [inputFocused, setInputFocused] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery), [setAppliedQuery]);
+
+  const handleSelection = (person: Person | null) => {
+    setSelectedPerson(person);
+
+    if (person) {
+      setQuery(person.name);
+      applyQuery(person.name);
+    }
+  };
+
+  const getFilteredPeople = useMemo(
+    () =>
+      [...peopleFromServer].filter(person =>
+        person.name.toLowerCase().includes(appliedQuery.trim().toLowerCase()),
```

Comment:
The filtering logic should handle inputs with only spaces. Consider trimming the input before checking if it includes the person's name to avoid unnecessary filtering.